src/daft-execution/src/ops/filter.rs

```diff
@@ -0,0 +1,49 @@
+use std::sync::Arc;
+
+use common_error::DaftResult;
+use daft_dsl::ExprRef;
+use daft_micropartition::MicroPartition;
+use daft_plan::ResourceRequest;
+
+use crate::partition::partition_ref::PartitionMetadata;
+
+use super::PartitionTaskOp;
+
+/// Filter task op, applying a predicate to its input.
+#[derive(Debug)]
+pub struct FilterOp {
+    predicate: Vec<ExprRef>,
+    resource_request: ResourceRequest,
+}
+
+impl FilterOp {
+    pub fn new(predicate: Vec<ExprRef>) -> Self {
+        Self {
+            predicate,
+            resource_request: ResourceRequest::new_internal(Some(1.0), None, None),
```

Comment:
Our current `Default::default()` is `None, None, None`, didn't want to change that since I think we rely on that elsewhere. I'll see if we can change the `Default` impl, otherwise we can add a `ResourceRequest::cpu_default()` to get rid of some of this boilerplate.